docs/developing-flows/flow-structure.md

```diff
@@ -0,0 +1,115 @@
+
+<div style="width: 300px" class="figure">
+  <img src="images/placeholder.png" alt="Link nodes">
+  <p class="caption">Placeholder for image: Link nodes</p>
+</div>
+
+
+There are some important differences between the two approaches. Link nodes cannot be used in the middle of a flow, where messages are passed over the link and then return when the other flow completes. They can only be used to start or end a flow. They can also be connect to more than one other link node. This lets you pass messages out to multiple other flows, or have multiple flows pass messages into a single flow.
+
+Subflows appear as regular nodes so can be used at any point in a flow. However each instance of the subflow is independent of the others. Any flow context inside the subflow will be scoped to the individual instances. If the subflow creates a connection to a remote system, each instance will create its own connection.
+
+### Customising subflows
+
+When creating subflows, you may want to be able to customise their behaviour in some way. For example, changing what MQTT topic it publishes to.
+
+
+One pattern for doing that is by setting the custom topic on the messages passed to the subflow. But that results in adding a Change node in front of every subflow instance in order to set the desired value.
+
+<div style="width: 300px" class="figure align-right">
+  <img src="images/placeholder.png" alt="MQTT topic set by environment variables">
+  <p class="caption">Placeholder for image: MQTT topic set by environment variables</p>
+
+  <img src="images/placeholder.png" alt="Adding subflow property">
+  <p class="caption">Placeholder for image: Adding subflow property</p>
+
+  <img src="images/placeholder.png" alt="Customising subflow instance property">
+  <p class="caption">Placeholder for image: Customising subflow instance property</p>
+
+</div>
+
+An easier way for doing this is by using Subflow properties. These are properties that can be set on the subflow instance and appear as environment variables inside the subflow.
+
+In the MQTT example, you could configure the node to publish to ${MY_TOPIC} and then add MY_TOPIC as a subflow property. When a user edits an individual instance they can then provide a custom value for MY_TOPIC.
+
+This pattern can be applied to any node configuration field that lets you enter the value directly. It doesn’t currently work for fields that are exposed as checkboxes or other custom UI elements.
+
+### Managing state information
+
+Another consideration is how to manage any state information in your flows. For example, keeping a count of how many messages pass through a flow, or the current state of an external sensor.
+
+Node-RED provides the Context system for managing state within the runtime. The context can be scoped to the same tab, subflow or made globally available.
+
```

Comment:
Should we also mention persistence over restart here as well  (or lack of by default) ? link to other docs etc ?

Reply:
I plan to do a separate pass through the doc adding links to relevant sections.